### ISSUE TITLE
The modal component needs relevant aria attributes to be properly interpreted by screen readers

### ISSUE THREAD

**Issue by nicklemmon** [SparkPost/matchbox#242]
## Requirements

* Modal dialogs should have `role="dialog"`
* Modal dialogs should *also* have `aria-modal="true"`

That's it! Pretty easy.

**Comment by jonambas** [comment]
Resolved in #244
